include/mqt-core/dd/Package.hpp

```diff
@@ -1462,8 +1440,7 @@ template <class Config> class Package {
       return {y.p, rWeight};
     }
 
-    if constexpr (std::is_same_v<RightOperandNode, mNode> ||
-                  std::is_same_v<RightOperandNode, dNode>) {
+    if constexpr (std::is_same_v<RightOperandNode, mNode>) {
```

Comment:
Fixing the density matrix multiplication failures might be as easy as reverting this change here.